service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/AwaitCloneAllResourcesFlightStep.java

```diff
@@ -53,37 +49,26 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         context.getInputParameters().get(JobMapKeys.REQUEST.getKeyName(), Workspace.class);
 
     try {
-      //noinspection deprecation
-      FlightState subflightState =
-          context
-              .getStairway()
-              .waitForFlight(
-                  cloneAllResourcesFlightId,
-                  FLIGHT_POLL_SECONDS,
-                  AWAIT_CLONE_RESOURCES_POLL_CYCLES);
-      if (FlightStatus.SUCCESS != subflightState.getFlightStatus()) {
+      StepResultWithFlightInfo subflightResult =
+          FlightUtils.waitForSubflightCompletionAndReturnFlightInfo(
+              context.getStairway(), cloneAllResourcesFlightId);
+
+      if (!subflightResult.isSuccess()) {
         // no point in retrying the await step
-        return new StepResult(
-            StepStatus.STEP_RESULT_FAILURE_FATAL,
-            subflightState
-                .getException()
-                .orElseGet(
-                    () ->
-                        new RuntimeException(
-                            String.format(
-                                "Subflight had unexpected status %s. No exception for subflight found.",
-                                subflightState.getFlightStatus()))));
+        return subflightResult.getStepResult();
```

Comment:
The constructed StepResult will include exception information.
https://github.com/DataBiosphere/terra-workspace-manager/pull/1688/files#diff-af300abcc49577c64d79724ef7af4288bbe77f0e6f5ed6c4e42e2a9a323379adR240